packages/drivers/form/misc/test-suite.js

```diff
@@ -0,0 +1,180 @@
+/* eslint-env jest */
+/* eslint-disable react/jsx-filename-extension */
+import { createStore as krmlCreateStore, when, types } from 'k-ramel'
+
+const defaultValues = { first: 'value', second: 'otherValue' }
+
+export default (driver) => {
+  const createStore = (listeners = []) => params => krmlCreateStore(
+    // TODO: krf doesn't make a reducer with a path. Only add on exist reducer.
+    { ui: { default: types.bool() } },
+    {
+      listeners,
+      drivers: {
+        form: driver(params),
+      },
+    },
+  )
+
+  const makeReaction = params => (reaction) => {
+    const store = createStore([
+      when('DISPATCHED')(reaction),
+    ])(params)
+
+    store.dispatch({ type: 'DISPATCHED' })
+    return store
+  }
+
+  const tests = (params) => {
+    it('should initialize', () => {
+      // store
+      const store = createStore()(params)
+      // assert
+      expect({
+        state: store.getState(),
+      }).toMatchSnapshot()
+    })
+
+    it('should init', () => {
+      const store = makeReaction(params)((action, st, { form }) => {
+        form('form').init()
+      })
+      // assert
+      expect({
+        state: store.getState(),
+      }).toMatchSnapshot()
+    })
+
+    it('should set values', () => {
+      const store = makeReaction(params)((action, st, { form }) => {
+        form('form').set(defaultValues)
+      })
+      // assert
+      expect({
+        state: store.getState(),
```

Comment:
From here, can you only retrieve data you need to test instead of all state (easier to maintain)